content/docs/command-reference/diff.md

```diff
@@ -33,6 +33,14 @@ sections below for more details.
 `dvc diff` does not have an effect when the repository is not tracked by Git,
 for example when `dvc init` was used with the `--no-scm` option.
 
+By default, when comparing the current workspace to a Git commit, `dvc diff`
+includes a `not in cache` status for DVC-tracked files and directories which are
+neither in the workspace nor the local cache (indicating that they most likely
+need to be `dvc pull`ed into the workspace). The `--hide-missing` option can be
+used to suppress this status and only print files and directories which have
+been expliclity added, modified or deleted in the workspace. This status will be
+automatically suppressed when comparing two Git commits.
```

Comment:
My 2cs on this - unless it's some critical option, I would try to make majority of this description into the Options section below (it's not scalable to describe every option in the main section, also the purpose of it is to describe general behavior, edge cases, etc. So, it's fine to reference an option, but probably not provide full description)